instrumentation/src/com/intellij/rt/coverage/instrumentation/data/BranchDataContainer.java

```diff
@@ -129,16 +133,33 @@ public void removeLastSwitch(Label dflt, Label... labels) {
     lineData.removeSwitch(lineData.switchesCount() - 1);
   }
 
+  public TIntArrayList getInstructions() {
+    return myInstructions;
+  }
+
+  public void addInstructions(int id, int instructions) {
+    myInstructions.set(id, myInstructions.get(id) + instructions);
+  }
+
+  private int getNextId() {
+    if (myInstructions != null) {
+      while (myInstructions.size() <= myNextId) {
+        myInstructions.add(0);
```

Comment:
write operation in getter, to my taste, kind of unexpected. I'd rename to `incId` if it would still work well together